src/server/rdb_load.cc

```diff
@@ -1957,6 +1957,9 @@ RdbLoader::~RdbLoader() {
 error_code RdbLoader::Load(io::Source* src) {
   CHECK(!src_ && src);
 
+  is_tiered_enabled_ =
+      shard_set->Await(0, [] { return EngineShard::tlocal()->tiered_storage() != nullptr; });
```

Comment:
Don't we have a flag for tiering, why not use that instead? Furthermore, I know we can start df with less shards than proactors. What I don't remember is if the shards start at index 0 or no for that case (where we have more proactors than shards). If they don't, then this is a bug here.